go-controller/pkg/ovn/service.go

```diff
@@ -110,6 +110,18 @@ func (ovn *Controller) syncServices(services []interface{}) {
 					svcRejectACLs[name] = make(map[string]bool)
 				}
 				svcRejectACLs[name][lb] = hasEndpoints
+
+				// Cloud load balancers: directly load balance that traffic from pods
+				for _, ing := range service.Status.LoadBalancer.Ingress {
+					if ing.IP == "" {
+						continue
+					}
+					name := ovn.generateACLName(lb, ing.IP, svcPort.Port)
+					if _, ok := svcRejectACLs[name]; !ok {
+						svcRejectACLs[name] = make(map[string]bool)
+					}
+					svcRejectACLs[name][lb] = hasEndpoints
```

Comment:
maybe add a helper for this since it's duplicated 4 times now

Reply:
yeah there are a bunch of patterns like this duplicated in this file. @alexanderConstantinescu already refactored it once as it was worse before. Perhaps we can take another stab at it:
https://github.com/ovn-org/ovn-kubernetes/issues/1828

Reply:
I pushed an fixup PR to tim's branch for this